release.md

```diff
@@ -1,30 +1,94 @@
+
+Completed releases are denoted by a _git tag_ and a corresponding [GitHub release](https://github.com/falcosecurity/libs/releases).
+
+## Release team
+
+The release team consists of a *release manager* and other contributors from the community. Usually, we seek volunteers during our [community calls](https://github.com/falcosecurity/community#community-calls) or in the #falco channel in slack](https://kubernetes.slack.com/messages/falco), and then decide together the release planning.
+
+The release manager's responsibility is to coordinate the release process and track its progress. 
+
+Note that the release manager does not need to be a maintainer. However, two [maintainers](https://github.com/falcosecurity/libs/blob/master/OWNERS) with repository rights must be part of the team to approve PRs, do the git tags, manage GitHub milestones, etc.
+
+## Release Phases
+
+Regardless if it is a driver or a libs release when initiating a new release, we follow a streamlined process inspired by the [Kubernetes Release Phases](https://github.com/kubernetes/sig-release/blob/master/releases/release_phases.md).
+
+![release-process-overview](docs/img/release-process.svg "Code Freeze to Thaw")
+
+### Preparation
+
+In the preparation phase, all enhancements expected to go into the release should be merged before the code freeze.
+
+When close to the code freeze starting period, maintainers should evaluate which PRs can't be merged in time and eventually postpone them to the subsequent milestone.
+
+It's recommended to start the preparation phase ~4 weeks before the release due date. 
+
+### Code freeze
+
+Code freeze happens ~1 week before the release due date, and should last no more than ~5 days. Shorter freeze period are encouraged.
+
+At this point, no new-feature PRs are allowed to be merged.
+
+[Versioning](#versioning) rules must be double-checked and eventually enforced at this stage.
+
+### Release branch
+
+During the code freeze period, a *release branch* (e.g., `release-x.y.z`) is created once the [release team](#release-team) ensures the code is in a good shape and reasonably no bugs are detected.
+
+The release branch name must include the release version. If the process consists of both drivers and libs releases, the libs one is used.
+
+In this phase:
+
+ - A PR must be opened in our [test-infra](https://github.com/falcosecurity/test-infra/blob/master/config/config.yaml) repository to set the newly created branch as protected.
+
+ - A release candidate tag should be created in the release branch for testing purposes.
+
+ - Accurate testing is performed on the release candidate. It's highly recommended to use Falco as a consumer when testing libs and drivers.
+
+ - If necessary, bug fixing or testing PRs can be merged on the `master` branch. In such a case, relevant commits are cherry-picked and ported to the release branch and a new release candidate is created.
```

Comment:
Similarly can `if necessary` be quantified?

Reply:
How? :smile_cat: 

I believe it's implicit for bug fixing (if one discovers a bug, then a fix is needed).

For testing, I assumed we might need to increase the code coverage in case we are not confident about a feature or a part of the code that will be released. Does it make sense?

Anyway, I'd not be too strict since the goal is to get a lean process.

Reply:
Perhaps adopt an approach similar to "vulnerability management"? ... and decide if it's worth the trouble now or do later without adding too many procedures either.

"Severe bugs that affect stability, safety, performance or feature bugs critical to libs core functionality can be merged on ..."

Reply:
done